### PR TITLE
Added the actual implementation for the `AtomicView::latest_view`

### PR DESCRIPTION
**Pull request by xgreenx**
Closes https://github.com/FuelLabs/fuel-core/issues/1581

The change finally implements an atomic view of the RocksDB database via the snapshot feature.

We use a snapshot (if available) each time we read/iterate over the RocksDB. It guarantees a consistent view even if the data is removed/added/modified.

`InMemory` database also implements this feature by simply cloning all `BTreeMap`s. In the future, we need to optimize this behavior and cache views - https://github.com/FuelLabs/fuel-core/issues/1993.

## Checklist
- [x] New behavior is reflected in tests

### Before requesting review
- [x] I have reviewed the code myself
- [x] I have created follow-up issues caused by this PR and linked them here

### CODE REVIEW

File: crates/fuel-core/src/state/in_memory/memory_store.rs [hunk 1]
```diff
@@ -62,6 +69,23 @@ impl<Description> MemoryStore<Description>
 where
     Description: DatabaseDescription,
 {
+    fn create_view(&self) -> MemoryView<Description> {
+        // Lock all tables at the same time to have consistent view.
+        let locks = self
+            .inner
+            .iter()
+            .map(|lock| lock.lock().expect("Poisoned lock"))
```
Comment (MitchTurner) [inline crates/fuel-core/src/state/in_memory/memory_store.rs:77]: Can we explain in the `expect` why we assume this isn't poisoned? I'm curious, but it's also best practice.

Comment (xgreenx) [reply crates/fuel-core/src/state/in_memory/memory_store.rs:77]: The code below works in the same way=D We could replace it with another lock that doesn't require expect in a separate RP.

Comment (MitchTurner) [reply crates/fuel-core/src/state/in_memory/memory_store.rs:77]: That would be even better than what I suggested.

**Review by MitchTurner — APPROVED** [review]
The tests really helped to understand a lot of the code.

**Pull request merged**